padl/transforms.py

```diff
@@ -1552,16 +1634,63 @@ def _pd_get_keys(transforms):
             deduped_keys.append(new_name)
         return deduped_keys
 
+    def _pd_get_non_target_stage_idx(self, stage_used: bool):
+        """When a :class:`Compose` gets an Exception and has a non-failing stage which is a
+        :class:`Parallel` or a :class:`Rollout`, the index they contribute to track the
+        element that got the Exception on the entire :class:`Compose` is 1.
+
+        Example:
+            t = ((prep_1 >> batch) + (prep_2 >> batch)) >> forward_1
+            Let's suppose we get an error on `forward_1`, so the item that is failing on `t` is 1.
+            `forward_1` is the index 0 of `t.pd_forward`, then the element that fails on `t` is
+            t._pd_preprocess._pd_get_non_target_stage_idx() + 0 = 1 + 0 = 1
```

Comment:
`t._pd_preprocess` to `t.pd_preprocess`, it is not a hidden method.

Reply:
changing it